src/lib.rs

```diff
@@ -1579,4 +1600,67 @@ mod test {
             r1 = core.poll_workflow_task().await.unwrap();
         }
     }
+
+    #[tokio::test]
+    async fn max_activites_respected() {
+        let mut tasks = VecDeque::from(vec![
+            PollActivityTaskQueueResponse {
+                task_token: vec![1],
+                activity_id: "act1".to_string(),
+                ..Default::default()
+            },
+            PollActivityTaskQueueResponse {
+                task_token: vec![2],
+                activity_id: "act2".to_string(),
+                ..Default::default()
+            },
+            PollActivityTaskQueueResponse {
+                task_token: vec![3],
+                activity_id: "act3".to_string(),
+                ..Default::default()
+            },
+        ]);
+        let mut mock_gateway = MockServerGatewayApis::new();
+        mock_gateway
+            .expect_poll_activity_task()
+            .times(3)
+            .returning(move || Ok(tasks.pop_front().unwrap()));
+        mock_gateway
+            .expect_complete_activity_task()
+            .returning(|_, _| Ok(RespondActivityTaskCompletedResponse::default()));
+
+        let core = CoreSDK::new(
+            mock_gateway,
+            CoreInitOptions {
+                gateway_opts: ServerGatewayOptions {
+                    target_url: Url::from_str("https://fake").unwrap(),
+                    namespace: "".to_string(),
+                    task_queue: "task_queue".to_string(),
+                    identity: "".to_string(),
+                    worker_binary_id: "".to_string(),
+                    long_poll_timeout: Default::default(),
+                },
+                evict_after_pending_cleared: true,
+                max_outstanding_workflow_tasks: 1,
+                max_outstanding_activities: 2,
+            },
+        );
+
+        // We allow two outstanding activities, therefore first two polls should return right away
+        let r1 = core.poll_activity_task().await.unwrap();
+        let _r2 = core.poll_activity_task().await.unwrap();
+        // Third should block until we complete one of the first two
```

Comment:
I don't understand how this is tested, how do we know the 3rd poll was blocked?

Reply:
Good point. Added a way to verify that